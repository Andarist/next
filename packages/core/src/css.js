// @flow
import type { CSSCache, Interpolation, ScopedInsertableStyles } from './types'
import { hashString } from 'emotion-utils'
import { handleInterpolation, labelPattern } from './serialize'

export function css(
  strings: Interpolation | string[],
  ...interpolations: Interpolation[]
): ScopedInsertableStyles & { toString: () => string } {
  let stringMode = true
  let styles: string = ''
  let identifierName = ''

  if (strings == null || strings.raw === undefined) {
    stringMode = false
    styles += handleInterpolation.call(this, strings)
  } else {
    styles += strings[0]
  }

  interpolations.forEach(function(interpolation, i) {
    styles += handleInterpolation.call(this, interpolation)
    if (stringMode === true && strings[i + 1] !== undefined) {
      styles += strings[i + 1]
    }
  }, this)
  styles = styles.replace(labelPattern, (match, p1: string) => {
    identifierName += `-${p1}`
    return ''
  })
  let name

  const ret: Object = {
    styles,
    type: 1,
    get name() {
      delete this.name
      this.name = hashString(styles) + identifierName
      return this.name
    },
    toString() {
      return `css-${this.name}`
    }
  }
  return ret
}
