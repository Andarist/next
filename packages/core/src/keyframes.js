// @flow
import * as React from 'react'
import { CSSContext } from './context'
import type { CSSContextType, InsertableStyles } from './types'
import { hydration, isBrowser, insertStyles } from './utils'
import { serializeStyles } from './serialize'

export const keyframes = (arg: { styles: string, name: string }) => {
  const name = `animation-${arg.name}`
  const ret = {
    styles: `@keyframes ${name}{${arg.styles}}`,
    name,
    scope: '',
    toString: () => `animation-${arg.name}`
  }
  return ret
}
