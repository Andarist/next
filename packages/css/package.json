{
  "name": "@emotion/css",
  "version": "0.2.3",
  "description": "a function to serialize css and object styless",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/next/tree/master/packages/css",
  "dependencies": {
    "@emotion/serialize": "^0.2.1",
    "@emotion/types": "^0.2.3",
    "emotion-utils": "^9.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist"
  ]
}
