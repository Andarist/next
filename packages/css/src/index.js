// @flow
import type { Interpolation, ScopedInsertableStyles } from '@emotion/types'
import hashString from '@emotion/hash'
import { handleInterpolation, labelPattern } from '@emotion/serialize'

function css(
  strings: Interpolation | string[],
  ...interpolations: Interpolation[]
): ScopedInsertableStyles {
  let stringMode = true
  let styles: string = ''
  let identifierName = ''

  if (strings == null || strings.raw === undefined) {
    stringMode = false
    styles += handleInterpolation.call(this, strings)
  } else {
    styles += strings[0]
  }

  interpolations.forEach(function(interpolation, i) {
    styles += handleInterpolation.call(this, interpolation)
    if (stringMode === true && strings[i + 1] !== undefined) {
      styles += strings[i + 1]
    }
  }, this)
  styles = styles.replace(labelPattern, (match, p1: string) => {
    identifierName += `-${p1}`
    return ''
  })
  let name = hashString(styles) + identifierName

  return {
    type: 1,
    name,
    styles
  }
}

export default css
