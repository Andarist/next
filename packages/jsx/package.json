{
  "name": "@emotion/jsx",
  "version": "0.0.5",
  "description": "a custom createElement for emotion with the css prop",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/next/tree/master/packages/jsx",
  "dependencies": {
    "@emotion/utils": "^0.0.5"
  },
  "peerDependencies": {
    "@emotion/core": "^0.0.4",
    "react": ">=16.3.0 || 16.3.0-alpha.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist"
  ]
}
