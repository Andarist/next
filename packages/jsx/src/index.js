// @flow
import * as React from 'react'
import { hydration, consumer } from '@emotion/core'
import { isBrowser, getRegisteredStyles, insertStyles } from '@emotion/utils'
import type { CSSContextType } from '@emotion/types'
import { serializeStyles } from '@emotion/serialize'

type Props = {
  props: Object | null,
  type: React.ElementType,
  children: Array<React.Node>,
  context: CSSContextType
}

class Style extends React.Component<Props> {
  shouldHydrate: boolean
  serialized: string
  constructor(props) {
    super(props)
    this.shouldHydrate = hydration.shouldHydrate
  }
  componentDidMount() {
    hydration.shouldHydrate = false
  }
  render() {
    const { props, type, children, context } = this.props
    let actualProps = props || {}

    let registeredStyles = []

    let className = ''
    if (actualProps.className !== undefined) {
      className = getRegisteredStyles(
        context.registered,
        registeredStyles,
        actualProps.className
      )
    }
    registeredStyles.push(
      typeof actualProps.css === 'function'
        ? actualProps.css(context.theme)
        : actualProps.css
    )
    const serialized = serializeStyles(registeredStyles)
    const rules = insertStyles(context, serialized)
    className += serialized.cls
    if (this.serialized === undefined && (this.shouldHydrate || !isBrowser)) {
      this.serialized = rules
    }

    const newProps = {
      ...actualProps,
      className
    }
    delete newProps.css
    const ele = React.createElement(type, newProps, ...children)
    if (this.serialized !== undefined) {
      return (
        <React.Fragment>
          <style
            data-more=""
            dangerouslySetInnerHTML={{ __html: this.serialized }}
          />
          {ele}
        </React.Fragment>
      )
    }
    return ele
  }
}

// $FlowFixMe
const jsx: typeof React.createElement = (
  type: React.ElementType,
  props: Object | null,
  ...children: Array<React.Node>
) => {
  if (props == null || props.css == null || type.__emotion_component === true) {
    return React.createElement(type, props, ...children)
  }
  if (process.env.NODE_ENV === 'development' && typeof props.css === 'string') {
    throw new Error(
      `Strings are not allowed as css prop values, please wrap it in a css template literal like this: css\`${
        props.css
      }\``
    )
  }

  return consumer(context => (
    <Style props={props} type={type} children={children} context={context} />
  ))
}

export default jsx
