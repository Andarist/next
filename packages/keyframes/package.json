{
  "name": "@emotion/keyframes",
  "version": "0.6.0",
  "description": "a function to generate",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "repository": "https://github.com/emotion-js/next/tree/master/packages/keyframes",
  "dependencies": {
    "@emotion/types": "^0.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "src",
    "dist"
  ],
  "umd:main": "./dist/index.min.js"
}