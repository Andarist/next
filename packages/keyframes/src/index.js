// @flow
import type { ScopedInsertableStyles, Keyframes } from '@emotion/utils'

const keyframes = (arg: ScopedInsertableStyles): Keyframes => {
  const name = `animation-${arg.name}`
  return {
    name,
    styles: `@keyframes ${name}{${arg.styles}}`
  }
}

export default keyframes
