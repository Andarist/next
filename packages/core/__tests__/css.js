// @flow
/** @jsx jsx */
import jsx from '@emotion/jsx'
import Style from '@emotion/style'
import Global from '@emotion/global'
import Provider from '@emotion/provider'
import css from '@emotion/css'
import keyframes from '@emotion/keyframes'
import renderer from 'react-test-renderer'

test('thing', () => {
  const tree = renderer.create(
    <div>
      <div css={{ display: 'flex' }}>something</div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

// skipping since this API will change
test.skip('css call to render', () => {
  const cls = css`
    color: green;
  `
  const tree = renderer.create(
    <div>
      <Style styles={cls} />
      <div className={cls.toString()}>something</div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

// skipping since this API will change
test.skip('css call to render with array as styles', () => {
  const cls = css`
    color: green;
  `
  const cls2 = css`
    color: hotpink;
  `
  const tree = renderer.create(
    <div css={{ color: 'yellow' }}>
      <Style styles={[cls, cls2]} />
      <Style styles={[cls]} />

      <div className={cls.toString()}>something</div>
      <div className={cls2.toString()}>anothor thing</div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('keyframes', () => {
  const animation = keyframes(css`
    from {
      color: green;
    }
    to {
      color: hotpink;
    }
  `)
  const tree = renderer.create(
    <div>
      <Style styles={animation} />
      <div
        css={css`
          animation: ${animation} ${true};
        `}
      >
        {animation.name}
      </div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('keyframes insert in css call', () => {
  const animation = keyframes(css`
    from {
      color: green;
    }
    to {
      color: hotpink;
    }
  `)
  const tree = renderer.create(
    <div>
      <div
        css={css`
          animation: ${animation.name};
          ${animation.styles};
        `}
      >
        {animation.name}
      </div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('global', () => {
  const tree = renderer.create(
    <div>
      <Global
        css={css`
          body {
            color: hotpink;
          }
        `}
      />
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('@font-face in global', () => {
  const tree = renderer.create(
    <div>
      <Global
        css={{
          '@font-face': {
            fontFamily: 'some-name'
          }
        }}
      />
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('functions get toStringed in css calls', () => {
  expect(
    css`
      ${() => {}};
    `
  ).toMatchSnapshot()
})

test('css call composition', () => {
  const first = css`
    color: hotpink;
  `
  const second = css({ ':hover': first })
  expect(second).toMatchSnapshot()
})

test('theming with the css prop', () => {
  const tree = renderer.create(
    <Provider theme={{ primary: 'hotpink' }}>
      <div css={theme => ({ color: theme.primary })} />
    </Provider>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('object with false', () => {
  const tree = renderer.create(
    <div>
      <div css={{ color: 'hotpink', display: false }}>something</div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('label in css call', () => {
  const tree = renderer.create(
    <div>
      <div
        css={css`
          color: hotpink;
          label: this-is-hotpink;
        `}
      >
        something
      </div>
    </div>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('string as css prop throws', () => {
  expect(() => {
    renderer.create(
      <div>
        <div
          css={`
            color: hotpink;
          `}
        >
          something
        </div>
      </div>
    )
  }).toThrowErrorMatchingSnapshot()
})
