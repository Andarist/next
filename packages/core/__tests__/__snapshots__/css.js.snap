// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@font-face in global 1`] = `
<div>
  <style>
    @font-face {
      font-family: some-name;
    }
  </style>
</div>
`;

exports[`css call composition 1`] = `
Object {
  "name": "d4uv9l",
  "styles": ":hover{
    color: hotpink;
  }",
  "toString": [Function],
  "type": 1,
}
`;

exports[`css call to render 1`] = `
<div>
  <style>
    .emotion-0 {
      color: green;
    }
  </style>
  <div
    className="emotion-0"
  >
    something
  </div>
</div>
`;

exports[`css call to render with array as styles 1`] = `
Array [
  <style>
    .emotion-0 {
      color: yellow;
    }
  </style>,
  <div
    className="emotion-0"
  >
    <style>
      .emotion-1 {
        color: green;
      }
      
      .emotion-2 {
        color: hotpink;
      }
    </style>
    <div
      className="emotion-1"
    >
      something
    </div>
    <div
      className="emotion-2"
    >
      anothor thing
    </div>
  </div>,
]
`;

exports[`dynamic 1`] = `
<div>
  <style>
    .emotion-0 {
      color: hotpink;
    }
  </style>
  <div
    className="emotion-0"
  />
</div>
`;

exports[`functions get toStringed in css calls 1`] = `
Object {
  "name": "5s8nl0",
  "styles": "
      function _ref() {};
    ",
  "toString": [Function],
  "type": 1,
}
`;

exports[`global 1`] = `
<div>
  <style>
    body {
      color: hotpink;
    }
  </style>
</div>
`;

exports[`keyframes 1`] = `
<div>
  <style>
    @-webkit-keyframes animation-1ozfi5a {
      from {
        color: green;
      }
    
      to {
        color: hotpink;
      }
    }
    
    @keyframes animation-1ozfi5a {
      from {
        color: green;
      }
    
      to {
        color: hotpink;
      }
    }
  </style>
  <style>
    .emotion-0 {
      -webkit-animation: animation-1ozfi5a;
      animation: animation-1ozfi5a;
    }
  </style>
  <div
    className="emotion-0"
  >
    animation-1ozfi5a
  </div>
</div>
`;

exports[`label in css call 1`] = `
<div>
  <style>
    .emotion-0 {
      color: hotpink;
    }
  </style>
  <div
    className="emotion-0"
  >
    something
  </div>
</div>
`;

exports[`object with false 1`] = `
<div>
  <style>
    .emotion-0 {
      color: hotpink;
    }
  </style>
  <div
    className="emotion-0"
  >
    something
  </div>
</div>
`;

exports[`string as css prop throws 1`] = `
"Strings are not allowed as css prop values, please wrap it in a css template literal from '@emotion/css' like this: css\`
            color: hotpink;
          \`"
`;

exports[`theming with the css prop 1`] = `
Array [
  <style>
    .emotion-0 {
      color: hotpink;
    }
  </style>,
  <div
    className="emotion-0"
  />,
]
`;

exports[`thing 1`] = `
<div>
  <style>
    .emotion-0 {
      display: -webkit-box;
      display: -webkit-flex;
      display: -ms-flexbox;
      display: flex;
    }
  </style>
  <div
    className="emotion-0"
  >
    something
  </div>
</div>
`;
